app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: any) -> any:
+    cached_list = {}
+
+    def inner(*args) -> any:
+        if args in cached_list:
+
+            cached_list[args] = "Getting from cache"
+            print("Getting from cache")
+
+            for item in args:
+                return item
+
+        cached_list[args] = "Calculating new result"
+        print("Calculating new result")
+
+        return func(*args)
```

Comment:
save result of execution

Reply:
I need save "func(*args)" before I return this?

Reply:
you need func(*args) if there is no stored result in your dictionary with execution results